pkgs/os-specific/linux/zfs/2_1.nix

```diff
@@ -9,6 +9,7 @@
 let
   stdenv' = if kernel == null then stdenv else kernel.stdenv;
 in
+lib.warn "zfs_2_1 is likely to become unsupported soon by upstream, and may be removed from NixOS 24.11, consider migrating to zfs_2_2"
```

Comment:
Note that this warning will be also shown for every user that does `nix-env -q`.

Reply:
Maybe we should just remove it instead?

Reply:
It also shows up for ofborg, so it needs to be done differently.